src/components/molecules/UserStatusManager/UserStatusManager.tsx

```diff
@@ -0,0 +1,105 @@
+import React, { useCallback, useMemo } from "react";
+import { Button } from "react-bootstrap";
+
+import { USER_STATUSES } from "settings";
+
+import { UserStatus } from "types/User";
+
+import { Checkbox } from "components/atoms/Checkbox";
+import { UserStatusPanel } from "./components/UserStatusPanel";
+
+import "./UserStatusManager.scss";
+
+const renderDefaultUserStatuses = USER_STATUSES.map((userStatus, index) => (
+  <UserStatusPanel
+    key={userStatus.status + index}
+    userStatus={userStatus}
+    disabled
+  />
+));
+
+export interface UserStatusManagerProps {
+  venueId?: string;
+  checked: boolean;
+  userStatuses: UserStatus[];
+  onCheck?: () => void;
+  onDelete?: (index: number) => void;
+  onAdd?: () => void;
+  onPickColor?: (color: string, index: number) => void;
+  onChangeInput?: (
+    value: React.FormEvent<HTMLInputElement>,
+    index: number
+  ) => void;
+}
+
+export const UserStatusManager: React.FC<UserStatusManagerProps> = ({
+  checked,
+  onCheck,
+  userStatuses,
+  onDelete,
+  onAdd,
+  onPickColor,
+  onChangeInput,
+}) => {
+  const deleteStatus = useCallback(
+    (index: number) => {
+      onDelete && onDelete(index);
+    },
+    [onDelete]
+  );
+
+  const changeInput = useCallback(
+    (value: React.FormEvent<HTMLInputElement>, index: number) => {
+      onChangeInput && onChangeInput(value, index);
+    },
+    [onChangeInput]
+  );
+
+  const pickColor = useCallback(
+    (color: string, index: number) => {
+      onPickColor && onPickColor(color, index);
+    },
+    [onPickColor]
+  );
+
+  const renderVenueUserStatuses = useMemo(
+    () =>
+      userStatuses.map((userStatus, index) => (
+        <UserStatusPanel
+          key={`panel-${index}`}
```

Comment:
We almost certainly don't want to be using `index` here.

Reply:
Changed in: https://github.com/sparkletown/sparkle/pull/1532/commits/0c1f9f07a5696848014d78ad2604de3c1ed2d194

Using only `userStatus` here is not possible because it changes on input and the key forces react to re-render and lose focus. It's using status + index now which should be enough.